src/main/java/com/aws/greengrass/shadowmanager/sync/IotDataPlaneClientFactory.java

```diff
@@ -92,7 +100,18 @@ private boolean validPath(Node node, String key) {
         return validString(node, key) && Files.exists(Paths.get(key));
     }
 
-    private void configureClient(DeviceConfiguration deviceConfiguration) {
+    @SuppressWarnings({"PMD.AvoidCatchingGenericException"})
+    private IotDataPlaneClient configureAndGetClient() throws IoTDataPlaneClientCreationException {
+        if (iotDataPlaneClient.get() != null) {
+            return iotDataPlaneClient.get();
+        }
```

Comment:
this is a race.

`client = iotDataPlaneClient.get()`
`if (client != null) return client`